AAP-4189 (#365)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/aap-on-azure/proc-peer-vwan-hub-to-aap.adoc

> [id="proc-peer-vwan-hub-to-aap"]

Reviewer: No `_{context}` in the ID?

= Peering a VWAN Hub to the {AAPonAzureNameShort} Network

> Before you can peer a VWAN with the {AAPonAzureNameShort} VNET, you must have connected a hub network and at least one spoke network to the hub network of the Azure VWAN that you intend to connect {AAPonAzureNameShort} to.

Reviewer: ```suggestion
Before you can peer a VWAN with the {AAPonAzureNameShort} VNET, you must connect a hub network, and at least one spoke network, to the hub network of the Azure VWAN to which you intend to connect {AAPonAzureNameShort}.
```

Reviewer: This reads like a prerequisite to me rather than an abstract. What do we mean by peering a VWAN hub to the network? What is the user motivation for completing this procedure?

Author: Peering is explained earlier in the doc. The AAP VNET is isolated when it's created, so it needs to be peered with other networks so that (1) it can be reached and (2) it can reach networks where you want to perform automation. VWAN is one of 3 options for peering.

> Before you can peer a VWAN with the {AAPonAzureNameShort} VNET, you must have connected a hub network and at least one spoke network to the hub network of the Azure VWAN that you intend to connect {AAPonAzureNameShort} to.

Reviewer 2: It  sounds a little odd to say 'connect to the thing you intend to connect to'. Perhaps 'that you intend to use'? Just a suggestion.

Author: Yes: this was a comment from the SME

> @ariordan-redhat Nuts, I jumped the gun again with the peer review. Sorry. I didn't notice the WIP label and just went for it after I got the email notification. I can close all my comments if you prefer.
> 
> Maybe we could remove that code owners thing and create a peer review group? That way, when we're ready for review, we can add the label and then tag the group or individual reviewers.

No problem. I forgot to remove the WIP tag.

Author: @ianf77 Would this work?

"Before peering, you must connect a hub network and at least one spoke network to the hub network of the Azure VWAN that you intend to connect {AAPonAzureNameShort} to."

I think the SME wanted to be completely clear that the required network connections needed to be made to the specific WAN that you're connecting AAP to.

Reviewer 2: It's a weird sentence but I can't really think of a  way round it if the 'intend to connnect to' is also necessary

.Prerequisites

> * A pre-configured Azure VWAN that is already tested and working.

Reviewer: Does "tested" not imply the VWAN is already working? Is there something more definite that can indicate this state, like the VWAN has a particular status?

* One or more of the following connections to the VWAN:

> ** A DMZ network that contains Azure virtual machines that users will use to access {AAPonAzureNameShort}.

Reviewer: ```suggestion
** A DMZ network that contains Azure virtual machines that provide user access to {AAPonAzureNameShort}.
```

> ** A DMZ network that contains an Azure virtual machine that can be used with SSH tunneling to access {AAPonAzureNameShort}.

Reviewer: ```suggestion
** A DMZ network with an Azure virtual machine that provides SSH access to {AAPonAzureNameShort}.
```

> ** A VPN or Direct Connect service to your local network that routes traffic from machines on your local network to {AAPonAzureNameShort}.

Reviewer: ```suggestion
** A VPN or Direct Connect service that routes traffic from your local network to {AAPonAzureNameShort}.
```

Reviewer: Not sure if this really changes meaning but "local network" is repeated in this sentence.

> Peering the {AAPonAzureNameShort} VNET to your VWAN follows the same approach as peering any other VNET or network with a VWAN.

Reviewer: I'm not sure this sentence is that informative or useful. Seems like fluff to me.

> 

Reviewer: Missing `.Procedure`

. Navigate to the *Virtual Network Connections* page for the VWAN that you want to peer with your {PlatformNameShort} instance.

> . Use the following settings to create a connection between the VWAN hub and your {PlatformNameShort} instance:

Reviewer: ```suggestion
. Create a connection between the VWAN hub and your {PlatformNameShort} instance with the following settings:
```

Reviewer: I often try to put the thing that the user is doing before the tools that they use to do it.

In this case, the user wants to create a connection not use the settings. Know what I mean?

** *Connection Name*: _<Ansible_Automation_Platform_connection_name>_
** *Hubs*: Select one or more VWAN hub networks that the managed application VNET will peer with.
** *Subscription*: Select the subscription where {AAPonAzureNameShort} has been deployed.
** *Resource group*: The managed resource group of the managed application. It will typically be prefixed with “mrg-”.
** *Virtual network*: The VNET of the managed application. There is only one VNET in the managed resource group.
** *Propagate to none*: _No_
** *Associate Route Table*: Select the default route table or the appropriate route table that your organization has configured for VWAN. 
** *Propagate to Route Tables*: Select one or more default route tables or the appropriate route table that your organization has configured for VWAN. 
** *Propagate to labels*: Select labels if your organization uses them.

> ** *Static routes*: Do not complete this field.

Reviewer: This is seems like a bit of a mix of reference and procedure. It looks like an enumeration of all the settings coupled with the action that users should take.

How about making each one a substep with a well-defined action?

.. Specify the _<Ansible_Automation_Platform_connection_name>_ with the *Connection Name* field.
.. From the *Hubs* field, select one or more ...
...
.. Do not enter any value in the *Static routes* field.

Author: I agree.  
The Azure team specifically didn't want a proper procedure because it describes the Azure UI, which Red Hat don't support.
So all the peering procedures are along the lines of "Navigate to the correct page and enter these values in the following fields in the form".

Author: @oraNod I'll make this change in another PR, as I'll have to update the other peering procedures to be consistent with this one. The PM has some updates for various procedures, so I'll run this change by him at the same time.

When network peering completes, traffic routes to and from {AAPonAzureNameShort} though your VWAN hub network.

> For further information on connecting a VNET to a VWAN, refer to link:https://docs.microsoft.com/en-us/azure/virtual-wan/howto-connect-vnet-hub[Connect a virtual network to a Virtual WAN hub - portal] in the Microsoft Azure Virtual WAN documentation.

Reviewer: ```suggestion
[role="_additional-resources"]
.Additional resources

* link:https://docs.microsoft.com/en-us/azure/virtual-wan/howto-connect-vnet-hub[Connect a virtual network to a Virtual WAN hub - portal] (Microsoft Azure Virtual WAN documentation)
```
